salt/metalk8s/volumes/prepared/init.sls

```diff
@@ -48,4 +48,11 @@ Provision backing storage for {{ volume }}:
     - require:
       - metalk8s_volumes: Format backing storage for {{ volume }}
 
+Update pillar after volume provisionning:
+  module.run:
+    - saltutil.refresh_pillar:
+      - wait: True
```

Comment:
To make this call (and the above) synchronous, you'll need to explicitly pass `async=False`.
For reference, see:
https://github.com/saltstack/salt/issues/20590
https://github.com/saltstack/salt/pull/50168

Reply:
Which makes me think there are some workarounds we could remove now that we upgraded to Salt 3000 (cc @TeddyAndrieux)

Reply:
Ho good catch, will add `async=False` then.

Reply:
Will revert this for now, seems not supported (or I got the syntax wrong):

```
      message: 'Salt job ''PrepareVolume'' failed with: ''saltutil.refresh_pillar''
        failed: refresh_pillar() got an unexpected keyword argument ''async'''
```

Reply:
No indeed my bad, it seems the `wait: True` is the most recent implementation.